docs/docs/routing.md

```diff
@@ -0,0 +1,238 @@
+
+The simplest way to implement Routing in a React application is with a top-level Component that chooses which content to render by URL. For example, a top-level `App` component may decide what to render based on [window.location.pathname](https://developer.mozilla.org/en-US/docs/Web/API/Location):
+
+```js
+function Menu() {
+  return (
+    <div>
+      <a href="/news">News</a>
+      <a href="/profile">Profile</a>
+    </div>
+  );
+}
+
+function App() {
+  const location = window.location;
+
+  let content;
+  if (location.pathname == '/news') {
+    content = <h1>News</h1>;
+  } else if (location.pathname == '/profile') {
+    content = <h1>Profile</h1>;
+  } else {
+    content = <h1>Not Found</h1>;
+  }
+  
+  return (
+    <div>
+      <Menu />
+      <div>{content}</div>
+    </div>
+  );
+}
+```
+
+But while this app will work, it has one major issue; it will reload the entire page every time a user clicks a link.
+
+## HTML5 History
+
+The HTML5 [History API](https://developer.mozilla.org/en/docs/Web/API/History) makes it possible to update the browser's URL without causing a page reload. It also emits browser events when the user clicks *Back* or *Forward*. By using this API, you can write applications that respond faster and behave as expected when *Back* or *Forward* are clicked.
+
+Single page applications will often make use of the following two parts of the History API:
+
+- The [popstate](https://developer.mozilla.org/en-US/docs/Web/Events/popstate) browser event, which is emitted when *Back* or *Forward* are clicked
+- The [History.pushState()](https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_pushState()_method) method, which allows the app to replace the contents of the URL bar without actually navigating
+
+## History objects
+
+The HTML5 History API is not without issues.
+
+Firstly, there are inconsistencies between how the different browsers use the `popstate` event. Some versions of Chrome and Safari will emit an event on page load while other versions won't.
+
+Secondly, navigation events are only emitted when they're caused by external stimuli such as the *Back* or *Forward* buttons. When the application itself updates the URL, nothing will happen. This means that the application will need to manually pass the new URL from the link that caused it to the top-level component that is handling routing.
+
+Typically, these issues are solved by creating a separate object which manages access to the HTML5 History API. This object will do two things:
+
+- It will normalise events between browsers
+- It will emit navigation events caused by both the application and external stimuli
+
+While it is certainly possible to create your own object to manage access to the History API, in this guide we'll use the popular [history](https://github.com/mjackson/history) package.
+
+```bash
+npm install history --save
+```
+
+As the History API is global, our object will also need to be accessible everywhere.
```

Comment:
Maybe add "For example, we can create a separate file and export the object from it, so that any other modules can later import it."